src/careamics/losses/lvae/losses.py

```diff
@@ -0,0 +1,374 @@
+        out_mean, _ = predictions.chunk(2, dim=1)
+    else:
+        out_mean = predictions
+
+    recons_loss_nm = -1 * likelihood_NM(out_mean, targets)[0].mean()
+    recons_loss_gm = -1 * likelihood_GM(predictions, targets)[0].mean()
+    recons_loss = denoise_weight * recons_loss_nm + split_weight * recons_loss_gm
+    return recons_loss
+
+
+def _get_weighted_likelihood(
+    ll, ch1_recons_w=1, ch2_recons_w=1
+):  # TODO what's this ? added defaults Thsi can be removed
+    """Each of the channels gets multiplied with a different weight."""
+    if ch1_recons_w == 1 and ch2_recons_w == 1:
+        return ll
+    # TODO Should these weights be removed?
+    assert ll.shape[1] == 2, "This function is only for 2 channel images"
+
+    mask1 = torch.zeros((len(ll), ll.shape[1], 1, 1), device=ll.device)
+    mask1[:, 0] = 1
+    mask2 = torch.zeros((len(ll), ll.shape[1], 1, 1), device=ll.device)
+    mask2[:, 1] = 1
+
+    return ll * mask1 * ch1_recons_w + ll * mask2 * ch2_recons_w
+
+
+def get_kl_divergence_loss_usplit(
+    topdown_layer_data_dict: dict[str, torch.Tensor]
+) -> torch.Tensor:
+    """Function to compute the KL divergence loss for uSplit."""
+    kl = torch.cat(
+        [kl_layer.unsqueeze(1) for kl_layer in topdown_layer_data_dict["kl"]], dim=1
+    )
+    # NOTE: kl.shape = (16,4) 16 is batch size. 4 is number of layers.
+    # Values are sum() and so are of the order 30000
+    # Example values: 30626.6758, 31028.8145, 29509.8809, 29945.4922, 28919.1875, 29075.2988
+
+    nlayers = kl.shape[1]
+    for i in range(nlayers):
+        # topdown_layer_data_dict['z'][2].shape[-3:] = 128 * 32 * 32
+        norm_factor = np.prod(topdown_layer_data_dict["z"][i].shape[-3:])
+        # if _restricted_kl:
+        #     pow = np.power(2,min(i + 1, _multiscale_count-1))
+        #     norm_factor /= pow * pow
+
+        kl[:, i] = kl[:, i] / norm_factor
+
+    kl_loss = free_bits_kl(kl, 0.0).mean()
+    return kl_loss
+
+
+def get_kl_divergence_loss(topdown_layer_data_dict, img_shape, kl_key="kl"):
+    # TODO explain all params!
+    """kl[i] for each i has length batch_size resulting kl shape: (batch_size, layers)"""
+    kl = torch.cat(
+        [kl_layer.unsqueeze(1) for kl_layer in topdown_layer_data_dict[kl_key]],
+        dim=1,
+    )
+
+    # As compared to uSplit kl divergence,
+    # more by a factor of 4 just because we do sum and not mean.
+    kl_loss = free_bits_kl(kl, 1).sum()  # TODO hardcoded free_bits 1
+    # NOTE: at each hierarchy, it is more by a factor of 128/i**2).
+    # 128/(2*2) = 32 (bottommost layer)
+    # 128/(4*4) = 8
+    # 128/(8*8) = 2
+    # 128/(16*16) = 0.5 (topmost layer)
+
+    # Normalize the KL-loss w.r.t. the  latent space
+    kl_loss = kl_loss / np.prod(img_shape)
+    return kl_loss
+
+
+# Loss Computations
+# mask = torch.isnan(target.reshape(len(x), -1)).all(dim=1)
+
+
+def musplit_loss(model_outputs, targets, loss_parameters) -> dict:
+    """Loss function for MuSplit.
+
```

Comment:
So I do think it would be better if all the parameters are passed explicitly here. Can you help me understand what problem the loss parameters data class is solving? When is it initialised and when are the parameters initially set? Are they in the configuration?